suite/tests/CMakeLists.txt

```diff
@@ -2121,6 +2121,12 @@ if (CLIENT_INTERFACE)
       "" "")
   endif (NOT ARM)
 
+  tobuild_appdll(client.blackbox client-interface/blackbox.c)
+  configure_DynamoRIO_client(client.blackbox.appdll)
+  tobuild_ci(client.blackbox client-interface/blackbox.c "" "" "")
+  target_link_libraries(client.blackbox.dll client.blackbox.appdll)
```

Comment:
But an appdll is an *app* dll.  It's imported by the app executable and loaded by the app loader.  Does this then result in two copies, one for the app and one private copy?  That doesn't seem ideal, and it's non-obvious.  Please add comments explaining the intent here.

I think it would be cleaner to not have this be an appdll.

Reply:
No, there's only 1 copy. I am linking the app dll to the client dll. I am just using the existing function that calls it "app dll". Just to understand, by "private" dll above or by "not app dll", what do you mean? It will still be a dll linked to the client dll yes?

Reply:
tobuild_appdll links the dll to the app executable (in most cases -- maybe it doesn't in certain cases where the app dynamically loads it, did not check the code), so I would expect two copies: one loaded by the app loader and one loaded by DR's private loader.  Hence an app dll and a private dll.

Reply:
See above.